app/controllers/products_controller.rb

```diff
@@ -21,11 +14,15 @@ def index
   def new
     # give them a shell and invite them to fill out the data.  Allows introspection into the object in the view!
     @product = Product.new
+    @categories = Category.all.order(:name).map{|c| [ c.name, c.id ] }
+
   end
 
   def create
     @product = Product.new(product_access_params[:product])
     #any validation?
+    @categories = Category.all.order(:name).map{|c| [ c.name, c.id ] }
+
```

Comment:
I see this same line in the new, create, and edit methods. This could be dried up by pulling out into a private method, `get_categories`

And then `before_action :get_categories, only: [:new, :create, :edit]`